monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/ScanObservable.scala

```diff
@@ -68,9 +77,13 @@ class ScanObservable[A,R](
             }
 
           def onComplete(): Unit =
-            if (!isDone) {
-              isDone = true
-              out.onComplete()
+            // `onNext` might require error reporting, so we need for it
+            // to actually finish before we can try to forward `onComplete`
+            lastAck.syncOnComplete { _ =>
```

Comment:
Here you need to do `firstAck.syncOnContinue`.

You also need the same logic in `onError`, because otherwise the last `onNext` streamed can become concurrent with the `onError` signal and that's against the protocol.